apps/frontend/src/libs/components/table/table.tsx

```diff
@@ -43,6 +62,20 @@ const Table = <T extends object>({
 				<tbody className={styles["table-body"]}>
 					{table.getRowModel().rows.map((row) => (
 						<tr className={styles["table-row"]} key={row.id}>
+							{onRowSelect && (
+								<td className={styles["table-data"]}>
+									<input
```

Comment:
We need a separate common component for checkbox and it should be controlled using useForm, similar to search field

Reply:
I've been working on it for quite some time and I really cannot figure this out. 
So table should have SelectRowCell and inside SelectRowCell should be Checkbox. Can you please explain this more?Because now for me it seems redundant.
Also, if i use useFormController in Checkbox component, then should the useAppForm for that be in Table component? And how do I handle multiple checkboxes as a form? I believe every checkbox should have its unique name to use in useController but how do I watch all them like in search component.
Sorry for too many questions, but I'm really stuck.

Reply:
@liza-veis, please check the comment above🙏

Reply:
> So table should have SelectRowCell and inside SelectRowCell should be Checkbox. Can you please explain this more?Because now for me it seems redundant.

Because if we add useForm, we may need to pass handlers or additional logic, etc., and to avoid overcomplicating the table component, it's better to move this to a separate component (not common but internal to the table component). This is a special cell, and for all other special cells, we will create a custom component, that's also why.

> Also, if i use useFormController in Checkbox component, then should the useAppForm for that be in Table component? And how do I handle multiple checkboxes as a form? I believe every checkbox should have its unique name to use in useController but how do I watch all them like in search component.
Sorry for too many questions, but I'm really stuck.

1. let's create useSelectedItems hook to handle checkboxes selection, something like  
```tsx
const useSelectedItems = <T>(items: T[]): UseSelectedItemsHook<T> => {
  const [selectedItems, setSelectedItems] = useState<T[]>(items);

  const handleAdd = (id: T): void => {
    setSelectedItems((prevState: T[]) => prevState.concat(id));
  };

  const handleRemove = (id: T): void => {
    setSelectedItems((prevState: T[]) => prevState.filter((it) => it !== id));
  };

  const handleToggle = (id: T): void => {
    selectedItems.includes(id) ? handleRemove(id) : handleAdd(id);
  };

  return { onToggle: handleToggle, items: selectedItems };
};
```
items is our selected items, onToggle is our onRowSelect

2. then if we have this select cell as separate component, we will need to pass onToggle and isChecked to it. And inside it we can have something like this
```
  const { control, errors, reset } = useAppForm({
    defaultValues: {
      [name]: isChecked,
    },
  });

  useEffect(() => {
    reset({ // or set value
      [name]: isChecked,
    });
  }, [name, isChecked]);
```
now we will be able to pass control, errors, and name to checkbox

Reply:
the name that we need to somehow pass here should be our userIds

Reply:
Thank you so much for your assistance! I hope the changes I've made are correct.